test/k8sT/Node.go

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Authors of Cilium
+
+package k8sTest
+
+import (
+	"fmt"
+
+	. "github.com/onsi/gomega"
+
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+)
+
+var _ = Describe("K8sNode", func() {
```

Comment:
Hm, do we really need an e2e test for this functionality? Couldn't we write unit tests to assert that `(u *CiliumNodeUpdater) updateCiliumNode` does the right thing? I don't think we need a full K8s cluster to validate the behavior of that function. Granted, that's assuming that we implement [1] and [2], as well as abstracting away the K8s client.

Reply:
I thought it would be a good way to test the full pipeline (update Node -> node watcher -> controller -> update CiliumNode -> read new object from API server), but if we improve unit tests we could just get rid of this and still test it indirectly in the egress gw e2e tests (where an update to a Node should trigger the reconciliation logic)